content-services/content-sources-backend
# Fixes 5023: api to fetch template repo configs

## Summary

Adds an endpoint to fetch the repo configs for a template

## Testing steps

1. Create a couple repos and let them snapshot
2. Create a template with those repos
3. Fetch the repo config files in the template via this endpoint: `GET /templates/:template_uuid/config.repo`
4. You should see a response similar to this:

```
[test1]
name=test1
baseurl=http://pulp.content:8081/pulp/content/cs-8997773847/templates/1708416b-d0c6-429a-88ba-2c3201c12f66/4a13e189-4b4f-46e8-af93-59f7e32b28ad
module_hotfixes=0
gpgcheck=1
repo_gpgcheck=0
enabled=1
gpgkey=http://pulp.content:8000/api/content-sources/v1.0/repository_gpg_key/4a13e189-4b4f-46e8-af93-59f7e32b28ad
sslclientcert=/etc/pki/consumer/cert.pem
sslclientkey=/etc/pki/consumer/key.pem

[ansible-2-for-rhel-8-x86_64-rpms]
name=Red Hat Ansible Engine 2 for RHEL 8 x86_64 (RPMs)
baseurl=http://pulp.content:8081/pulp/content/cs-redhat/templates/1708416b-d0c6-429a-88ba-2c3201c12f66/content/dist/layered/rhel8/x86_64/ansible/2/os/
module_hotfixes=0
gpgcheck=1
repo_gpgcheck=0
enabled=1
gpgkey=http://pulp.content:8000/api/content-sources/v1.0/repository_gpg_key/3b487063-79ec-4226-b91f-4982a6192ec5
sslclientcert=/etc/pki/consumer/cert.pem
sslclientkey=/etc/pki/consumer/key.pem
```

5. Verify the distribution paths are correct
6. Verify that a client and normal users can access the endpoint. You can set your identity header to be of type System to test the client auth locally: `{"identity":{"type":"System","system":{},"account_number":"11491786","internal":{"org_id":"17684632"}}}`
